### PR TITLE
implemented PolicyMakings controller and associated views

### PR DESCRIPTION
**Pull request by baptjs**

### CODE REVIEW

File: app/assets/stylesheets/style.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/Users/baptistejosse/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/sass-3.7.4/lib/sass/plugin.rb:109:in `method_missing'
+/Users/baptistejosse/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/sass-3.7.4/lib/sass/exec/sass_scss.rb:358:in `watch_or_update'
+/Users/baptistejosse/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/sass-3.7.4/lib/sass/exec/sass_scss.rb:51:in `process_result'
+/Users/baptistejosse/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/sass-3.7.4/lib/sass/exec/base.rb:50:in `parse'
+/Users/baptistejosse/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/sass-3.7.4/lib/sass/exec/base.rb:18:in `parse!'
+/Users/baptistejosse/.rbenv/versions/2.7.3/lib/ruby/gems/2.7.0/gems/sass-3.7.4/bin/sass:13:in `<top (required)>'
+/Users/baptistejosse/.rbenv/versions/2.7.3/bin/sass:23:in `load'
+/Users/baptistejosse/.rbenv/versions/2.7.3/bin/sass:23:in `<main>'
+*/
+body:before {
```
Comment (arieivs) [inline app/assets/stylesheets/style.css]: what is this?

File: app/controllers/policy_makings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  skip_before_action :authenticate_user!, only: [:show]
+
+  def index
+    @policy_makings = PolicyMaking.all
+  end
+
+  def show
+    if params[:country_topic].present? # Request comes from the search bar (home page)
+      if params[:country_topic][:country].empty? || params[:country_topic][:topic].empty?
+        render 'pages/home' # Should we display error messages ?
```
Comment (arieivs) [inline app/controllers/policy_makings_controller.rb]: I think it would be a good idea to render an alert asking them to choose a country and topic, yes!

File: app/controllers/policy_makings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  private
+
+  def set_policy_making
+    @policy_making = PolicyMaking.find(params[:id])
+  end
+
+  def policy_making_params
+    params.require(:policy_making).permit(:country_id, :topic_id, :content, :ambassador_id)
```
Comment (arieivs) [inline app/controllers/policy_makings_controller.rb]: what about the video_url, alt_text and source? shouldn't we permit them as well?

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -4,6 +4,14 @@
   <%= f.error_notification %>
 
   <div class="form-inputs">
+    <%= f.input :first_name,
+                required: true,
+                autofocus: true,
```
Comment (arieivs) [inline app/views/devise/registrations/new.html.erb:9]: I think you should only have autofocus once, in the first item, else it's a bit the same as not having it at all :p not actually sure how it reacts with autofocus everywhere

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -13,9 +13,9 @@
   </head>
 
   <body>
-    <% if !current_page?(controller: 'pages', action: 'home') %>
-      <%= render 'shared/navbar' %>
-    <% end %>
+
```
Comment (arieivs) [inline app/views/layouts/application.html.erb:15]: why did you remove this?

File: app/views/pages/home.html.erb [hunk 6]
```diff
@@ -1,10 +1,25 @@
 <header class="header-banner">
-  <%= render 'shared/navbar' %>
+
```
Comment (arieivs) [inline app/views/pages/home.html.erb]: why did you remove the navbar? we still need a navbar on the home page...

File: app/views/pages/home.html.erb [hunk 7]
```diff
@@ -1,10 +1,25 @@
...
+        <div class="country-topic">
+            <div id="country-filter">
+              <%= f.input :country, required: true, collection: Country.all, input_html: {class: "select2"} %>
+            </div>
+            <div id="topic-filter">
+              <%= f.input :topic, required: true, collection: Topic.all, input_html: {class: "select2"} %>
+            </div>
+          </div>
+          <%= f.submit "Explore",formaction: policy_making_path(1), class: "btn-all btn-yllw" %>
+          <%= f.submit "Engage", formaction: about_path, class: "btn-all btn-yllw" %>
```
Comment (arieivs) [inline app/views/pages/home.html.erb]: I think the Engage button should be outside of the form, as it is separate and not related to the country and topic selection

File: app/views/policy_makings/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+<p> Policy Making id : <%= @policy_making.id %> </p>
+<p> Country: <%= @policy_making.country.name %> </p>
+<p> Topic: <%= @policy_making.topic.name %> </p>
+
+<%= simple_form_for @policy_making do |f| %>
+  <%= f.input :content, as: :text, label: "New content", required: false, :input_html => { :cols => 5, :rows => 10 } %>
+  <%= f.input :ambassador_id, label: "New ambassador", required: false, collection: Ambassador.all, input_html: {class: "select2"} %>
+
+  <div class="btn-container"><%= f.button :submit, "Update", class: 'btn-all btn-yllw' %></div>
```
Comment (arieivs) [inline app/views/policy_makings/edit.html.erb:11]: what is this btn-container class? didn't see it in styling nor couldn't find it on bootstrap. why is it necessary to put the submit button inside a div?

File: app/views/policy_makings/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        <td><%= policy_making.topic.name %></td>
+        <td><%= policy_making.ambassador.name %></td>
+        <td><%= link_to ('<i class="far fa-edit"></i>').html_safe, edit_policy_making_path(policy_making) %></td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+
+
+<style>
```
Comment (arieivs) [inline app/views/policy_makings/index.html.erb]: move this to a _table.scss file in assets/stylesheets/components

File: app/views/pages/home.html.erb [hunk 10]
```diff
@@ -1,10 +1,25 @@
...
   <div class="container banner">
     <div class="container-half">
       <h1>OpenPolicy Europe</h1>
       <p>The digital hub for understanding and engaging with policymaking</p>
       <a href="join.html"><button class="btn-all btn-yllw">Join us</button></a>
+
+      <%= simple_form_for :country_topic, method: :get do |f| %>
+        <div class="country-topic">
+            <div id="country-filter">
+              <%= f.input :country, required: true, collection: Country.all, input_html: {class: "select2"} %>
```
Comment (arieivs) [inline app/views/pages/home.html.erb]: what does select2 mean? (out of curiosity)

File: app/views/policy_makings/edit.html.erb [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<h1>PolicyMakings#edit</h1>
+
+<p> Policy Making id : <%= @policy_making.id %> </p>
```
Comment (arieivs) [inline app/views/policy_makings/edit.html.erb:3]: I don't think the id will be very relevant to the user, do you?

File: db/seeds.rb [hunk 12]
```diff
@@ -8,12 +8,71 @@
...
+
+# --------------- COUNTRIES ---------------
+
+puts "Cleaning COUNTRIES database..."
+Country.destroy_all
+
+puts "Creating countries..."
+COUNTRIES = ['Europe', 'Austria', 'Belgium', 'Bulgaria', 'Croatia', 'Republic of Cyprus', 'Czech Republic', 'Denmark', 'Estonia', 'Finland', 'France', 'Germany', 'Greece', 'Hungary', 'Ireland', 'Italy', 'Latvia', 'Lithuania', 'Luxembourg', 'Malta', 'Netherlands', 'Poland', 'Portugal', 'Romania', 'Slovakia', 'Slovenia', 'Spain', 'Sweden']
+COUNTRIES.each do |country|
+  Country.new(name: country).save # Add flags ?
```
Comment (arieivs) [inline db/seeds.rb:20]: I think we should add flags yes, question is, how? could we add it as an emoji? 😅 on a string? that would be nice. Else, as a picture I guess. If we add as a picture, better not to use all the countries for the seeds (after all, in reality we will only have Europe, Germany, Portugal and Italy to start with), as adding photos really increases the time it takes to seed

File: db/seeds.rb [hunk 13]
```diff
@@ -8,12 +8,71 @@
...
+
+# --------------- POLICY MAKINGS ---------------
+
+puts "Cleaning POLICY_MAKINGS database..."
+PolicyMaking.destroy_all
+
+puts "Creating policy makings..."
+PolicyMaking.new(country_id: 1, topic_id: 1, ambassador_id: 1, content: "blah blah blah").save
+PolicyMaking.new(country_id: 1, topic_id: 2, ambassador_id: 1, content: "blah blah blah").save
+PolicyMaking.new(country_id: 1, topic_id: 1, ambassador_id: 3, content: "blah blah blah").save
```
Comment (arieivs) [inline db/seeds.rb]: you have country 1 and topic 1 twice

File: app/assets/stylesheets/style.css (hunk 1 above)
Comment (baptjs) [reply app/assets/stylesheets/style.css]: I have no idea, this was added automatically... I'll delete it

File: app/controllers/policy_makings_controller.rb (hunk 2 above)
Comment (baptjs) [reply app/controllers/policy_makings_controller.rb]: Ok, I'll do that

File: app/controllers/policy_makings_controller.rb (hunk 3 above)
Comment (baptjs) [reply app/controllers/policy_makings_controller.rb]: Sure, I didn't use them yet but they'll have to be permitted of course

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (baptjs) [reply app/views/layouts/application.html.erb:15]: Because, here you render the navbar for all pages EXCEPT home, then on the home page view, you render the navbar... That looks weird. Was there a particular reason for that ?

File: app/views/pages/home.html.erb (hunk 6 above)
Comment (baptjs) [reply app/views/pages/home.html.erb]: There's no need to render it here since I removed the weird condition in application.html.erb
The navbar is already rendered on all pages

File: app/views/pages/home.html.erb (hunk 7 above)
Comment (baptjs) [reply app/views/pages/home.html.erb]: Really, it's not ? I thought that the Engage button would redirect to the opportunities related to the country/topic selected. If not, it's weird to have the Explore and Engage buttons on the same level just below the search bar... Really confusing

File: app/views/policy_makings/edit.html.erb (hunk 8 above)
Comment (baptjs) [reply app/views/policy_makings/edit.html.erb:11]: None of them is necessary... I actually copied this part from an old project of mine, but I should've cleaned it. I'll do it.

File: app/views/policy_makings/index.html.erb (hunk 9 above)
Comment (baptjs) [reply app/views/policy_makings/index.html.erb]: Yes, this is temporary of course, I just started the front end here and didn't have the time to finish. I will move it to a component.

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (arieivs) [reply app/views/layouts/application.html.erb:15]: Yes, because on the home page the navbar is inside the header, in order to have the picture on the background, and in this way it appears before the header

File: app/views/pages/home.html.erb (hunk 7 above)
Comment (arieivs) [reply app/views/pages/home.html.erb]: They're not at the same level anymore, the Explore button is supposed to be in front of the search bar, and the Engage button below, with some space and text in between

Comment (arieivs) [reply app/views/pages/home.html.erb]: Maybe you were looking for the Components & old stuff page on figma? that would also explain the navbar confusion

Comment (baptjs) [reply app/views/pages/home.html.erb]: Yes... 😒😒😒
It makes more sense now and there's a lot more material in Figma

File: app/views/policy_makings/edit.html.erb (hunk 11 above)
Comment (baptjs) [reply app/views/policy_makings/edit.html.erb:3]: Nope, this view was just here for debugging, I didn't do the front end yet

File: app/controllers/pages_controller.rb [hunk 14]
```diff
@@ -1,7 +1,25 @@
 class PagesController < ApplicationController
   skip_before_action :authenticate_user!, only: [:home, :about]
 
-  def home; end
+  def home
+    @policy_making = PolicyMaking.new
```
Comment (arieivs) [inline app/controllers/pages_controller.rb:5]: I'm curious: why making a new PolicyMaking here and why assign it the topic and country later on? This @policy_making seems to only be relevant when there's an error of some sort, but I don't fully understand why we need it sorry.
Note: what about moving the PolicyMaking.new to after the if params[:policy_making].present? so that nothing really happens when no params are there?

**Comment by baptjs** [comment]
You need it because of the simple_form_for @policy_making. It requires an instance of PolicyMaking to store the data that will be selected on the client side (country and topic). We will never save this instance in the db of course, but the data needs to be saved so that we can display it when we render the page again with error messages (if the "user" has selected a country but not a topic for example). It's exactly like for "new" methods in controllers, we instantiate the class just store the data temporarily (then in the "create" method, we instantiate the class AGAIN, and this time we do save that instance in the db).

Maybe there's a better way of doing it ? We can ask someone more experienced :) But at least it works !

**Comment by arieivs** [comment]
Got it, I did not think about the simple form. Thank you for the explanation 🙏

**Review by arieivs — APPROVED** [review]
